cmd/version.go

```diff
@@ -47,10 +49,54 @@ type version struct {
 
 // versionString returns the CLI version
 func versionString() string {
-	if kubeBuilderVersion == unknown {
-		if info, ok := debug.ReadBuildInfo(); ok && info.Main.Version != "" {
+	info, ok := debug.ReadBuildInfo()
+
+	if ok && info.Main.Version != "" {
+		if kubeBuilderVersion == unknown {
 			kubeBuilderVersion = info.Main.Version
 		}
+
+		if goos == unknown {
+			goos = runtime.GOOS
+		}
+
+		if goarch == unknown {
+			goarch = runtime.GOARCH
+		}
+
+		if gitCommit == unknown && info.Main.Version != "" {
+			mainVersionSplit := strings.Split(info.Main.Version, "-")
```

Comment:
@migueleliasweb wdyt about to utilize a semver parsing library which supports segments, instead of custom code?